qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/ModelIT.java

```diff
@@ -0,0 +1,232 @@
+            createKnnIndex(testIndex, modelIndexMapping(TEST_FIELD, TEST_MODEL_ID));
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, DOC_ID, NUM_DOCS);
+        } else {
+            DOC_ID = NUM_DOCS;
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, DOC_ID, NUM_DOCS);
+            QUERY_COUNT = 2 * NUM_DOCS;
+            validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, QUERY_COUNT, K);
+
+            searchKNNModel(TEST_MODEL_ID);
+
+            createKnnIndex(TEST_MODEL_INDEX, modelIndexMapping(TEST_FIELD, TEST_MODEL_ID));
+            addKNNDocs(TEST_MODEL_INDEX, TEST_FIELD, DIMENSIONS, DOC_ID_TEST_MODEL_INDEX, NUM_DOCS_TEST_MODEL_INDEX);
+            QUERY_COUNT_TEST_MODEL_INDEX = NUM_DOCS_TEST_MODEL_INDEX;
+            validateKNNSearch(TEST_MODEL_INDEX, TEST_FIELD, DIMENSIONS, QUERY_COUNT_TEST_MODEL_INDEX, K);
+
+            deleteKNNIndex(testIndex);
+            deleteKNNIndex(TRAINING_INDEX);
+            deleteKNNIndex(TEST_MODEL_INDEX);
+        }
+    }
+
+    // KNN model test Default Parameters
+    public void testKNNModelDefault() throws Exception {
+        if (isRunningAgainstOldCluster()) {
+
+            // Create a training index and randomly ingest data into it
+            createBasicKnnIndex(TRAINING_INDEX_DEFAULT, TRAINING_FIELD, DIMENSIONS);
+            bulkIngestRandomVectors(TRAINING_INDEX_DEFAULT, TRAINING_FIELD, NUM_DOCS, DIMENSIONS);
+
+            trainKNNModel(TEST_MODEL_ID_DEFAULT, TRAINING_INDEX_DEFAULT, TRAINING_FIELD, DIMENSIONS, MODEL_DESCRIPTION);
+            validateModelCreated(TEST_MODEL_ID_DEFAULT);
+
+            createKnnIndex(testIndex, modelIndexMapping(TEST_FIELD, TEST_MODEL_ID_DEFAULT));
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, DOC_ID, NUM_DOCS);
+        } else {
+            DOC_ID = NUM_DOCS;
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, DOC_ID, NUM_DOCS);
+            QUERY_COUNT = 2 * NUM_DOCS;
+            validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, QUERY_COUNT, K);
+
+            searchKNNModel(TEST_MODEL_ID);
+
+            createKnnIndex(TEST_MODEL_INDEX_DEFAULT, modelIndexMapping(TEST_FIELD, TEST_MODEL_ID_DEFAULT));
+            addKNNDocs(
+                TEST_MODEL_INDEX_DEFAULT,
+                TEST_FIELD,
+                DIMENSIONS,
+                DOC_ID_TEST_MODEL_INDEX_DEFAULT,
+                NUM_DOCS_TEST_MODEL_INDEX_DEFAULT
+            );
+            QUERY_COUNT_TEST_MODEL_INDEX_DEFAULT = NUM_DOCS_TEST_MODEL_INDEX_DEFAULT;
+            validateKNNSearch(TEST_MODEL_INDEX_DEFAULT, TEST_FIELD, DIMENSIONS, QUERY_COUNT_TEST_MODEL_INDEX_DEFAULT, K);
+
+            deleteKNNIndex(testIndex);
+            deleteKNNIndex(TRAINING_INDEX_DEFAULT);
+            deleteKNNIndex(TEST_MODEL_INDEX_DEFAULT);
+        }
+    }
+
+    // Delete ".opensearch-knn-models" index to delete all models
+    @AfterClass
+    public static void deleteKNNModelsIndex() throws IOException {
+        if (!isRunningAgainstOldCluster()) {
```

Comment:
If the purpose of this is to delete the models, we will need to use the DELETE models API which will clear out cluster metadata as well.